mysql-test/suite/sys_vars/t/net_buffer_shrink_interval_basic.test

```diff
@@ -0,0 +1,122 @@
+EFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEFEF");
+--echo see con2 buffer after grow
+show status like 'net_buffer_length';
+--echo See total usage after buffer has been grown
+show global status like 'net_buffer_length';
+
+--echo Wait for first shrink check interval expire
+SELECT SLEEP(3);
+
+--echo Shrink check triggered by queries, so trigger it in both threads
+show status like 'net_buffer_length';
+connection con_int1;
+
+show status like 'net_buffer_length';
+connection con_int2;
+
+--echo Wait for second shrink check interval expire
+SELECT SLEEP(3);
+
+--echo First query to trigger check, second to see shrinked buffer
+show status like 'net_buffer_length';
+show status like 'net_buffer_length';
+
+connection con_int1;
+
+--echo First query to trigger check, second to see shrinked buffer
+show status like 'net_buffer_length';
+show status like 'net_buffer_length';
+
+--echo See what is the global usage after both buffers have bee shrinked
+show global status like 'net_buffer_length';
+
+connection default;
+
+--echo Check default connections size
+show status like 'net_buffer_length';
+
+disconnect con_int1;
+disconnect con_int2;
+
+--echo "This also serves as a delay to give time for above disconnects to finish"
```

Comment:
The disconnect finish can be ensured by include/count_sessions.inc include/wait_until_count_sessions.inc pair